### ISSUE TITLE
Packages: monitoring readiness for GAP 4.11 

### ISSUE THREAD

**Issue by alex-konovalov** [gap-system/gap#2984]
This is an issue to monitor the status of selected packages for which an update is desired in time for GAP 4.11. This replaces #2804 and lists everything from there which was not ready by the time of GAP 4.10. Of course, some of these issues may be addressed in package releases included in GAP 4.10 series. Please feel free to edit this issue and add more packages to the checklist. 

**HIGHLY DESIRABLE**
- [x] AtlasRep to include the change already made by @ThomasBreuer as stated in https://github.com/gap-system/gap/issues/2789 (which will allow to merge #2791)
- [x] Resolve #2703 to fix the problem with CTblLib reported in #1619 (or make the change in CTblLib)
- [ ] GBNP
- [x] GRAPE to make it work under 64-bit GAP for Windows : https://github.com/gap-packages/grape/issues/10
- [x] Guarana: https://github.com/gap-packages/guarana/issues/4
- [x] LOOPS and FR: resolve https://github.com/gap-packages/loops/issues/4, currently having a segfault in a package tests when all packages a loaded, apparently related to https://github.com/gap-packages/fr/issues/28
- [x] ResClasses: https://github.com/gap-packages/resclasses/issues/2
- [x] [SmallGrp](https://github.com/gap-packages/smallgrp/): merge PR https://github.com/gap-packages/smallgrp/pull/46 to fix https://github.com/gap-packages/smallgrp/issues/38
- [x] [smallsemi](https://github.com/gap-packages/smallsemi/): new release from GitHub fixing https://github.com/gap-packages/smallsemi/issues/2 
- [x] [spinsym](https://github.com/gap-packages/spinsym) now is on GitHub, and a new release was made there
- [x] [tomlib](https://github.com/gap-packages/tomlib/): new release after merging https://github.com/gap-packages/tomlib/pull/28

**GOOD TO HAVE**
- [x] design: https://github.com/gap-packages/design/issues/1
- [x] format: https://github.com/gap-packages/format/issues/2
- [x] [YangBaxter](https://github.com/gap-packages/YangBaxter) to be suggested for the redistribution (see https://github.com/gap-packages/YangBaxter/issues/43)
- [ ] More fixes to make tests pass for packages from [staging tests](https://travis-ci.org/gap-system/gap-docker-pkg-tests-stable-4.10-staging):
  - [ ] atlasrep: diffs reported and fixes suggested in #1617
  - [x] ctbllib: diffs reported in #1619
  - [ ] FR to fix #3418
  - [ ] gbnp: @jwknopper contacted regarding the new release (including new repository and webpage)
  - [ ] hapcryst: reported to the author by @alex-konovalov in October 2018 (CC GAP Support)
  - [x] loops: as in the mentioned above issue https://github.com/gap-packages/loops/issues/4
  - [ ] polymaking: check if it is usable, and can it be  superseded by https://github.com/homalg-project/homalg_project/tree/master/PolymakeInterface ?
- [x] some packages currently submitted for the refereeing 

**PULL REQUESTS BLOCKED BY PACKAGES**
- [x] #2237 waits for atlasrep, gbnp. Done in hecke (https://github.com/gap-packages/hecke/issues/5) and xgap (https://github.com/gap-packages/xgap/issues/7)
- [x] #2791 waits for atlasrep
- [ ] #3006 waits for Semigroups, GUAVA. Done in numericalspgs.
- [ ] #2997 waits for NormalizInterface. 

**WINDOWS ISSUES**
Need also to check that as many as possible packages with binaries are usable under Windows.
- [x] curlInterface: see https://github.com/gap-packages/curlInterface/issues/13
- [x] Digraphs: see https://github.com/gap-packages/Digraphs/issues/138 and https://github.com/gap-packages/Semigroups/issues/480
- [ ] JupyterKernel https://github.com/gap-packages/JupyterKernel/issues/45
- [x] Semigroups: see https://github.com/gap-packages/Semigroups/issues/480

**ESTABLISHING PUBLIC REPOSITORIES**

The list of remaining packages which are redistributed with GAP but do not have known (to us) public repository is at https://github.com/gap-packages/gap-packages.github.io/issues/7. Please help to reduce it - by letting us know if the repository exists, of if you're a package author, by establishing it. We offer help with this:
- we can prepare for you a repository with revisions history populated from the past releases of your package
- we can automate publishing releases for you, and 
- we can run tests of the **development** version of your package, not only of its latest release.

**Comment by alex-konovalov** [comment]
Ticked the box for LOOPS and FR.

**Comment by wilfwilson** [comment]
Is there anything that the Semigroups/Digraphs authors can do to make it more likely that Semigroups/Digraphs will work with the new Windows release?

**Comment by wilfwilson** [comment]
Thanks!

**Comment by alex-konovalov** [comment]
It would be good to reduce a number of warnings displayed while package loading, and check that none of them point to serious issues:
```
#W  PackageDoc.Archive is withdrawn, use PackageDoc.ArchiveURLSubset instead
#I  method installed for IsSubset matches more than one declaration
#I  polymake command not found. Please set POLYMAKE_COMMAND by hand
#I  method installed for Matrix matches more than one declaration
#I  HAP warning: Set POLYMAKE_PATH manually if needed. 
#I  HAP warning: Set ASY_PATH manually if needed.
#I  Loaded interface to Normaliz (NormalizInterface)
#I  Loaded interface to 4ti2 (4ti2Interface)
#I  Loaded interface to Singular (Singular)
#I  equal requirements in multiple declarations for operation `AlgebraHomomorphismByFunction'

#I  method installed for IsSubset matches more than one declaration
#I  You may wish to install the xgap package
#I  and enjoy the graphic capabilities of SONATA.
#I  Package `itc' needs package `xgap' version at least 4.02
#I  method installed for [] matches more than one declaration
#I  equal requirements in multiple declarations for operation `ProductIdeal'

#I  method installed for PreImagesSet matches more than one declaration
#I  method installed for Factorization matches more than one declaration
#I  method installed for Factorization matches more than one declaration
#I  method installed for DirectProductOp matches more than one declaration
#I  method installed for SmallGeneratingSet matches more than one declaration
#I  method installed for Factorization matches more than one declaration
#I  method installed for StructureDescription matches more than one declaration
#I  method installed for StructureDescription matches more than one declaration
#I  simpcomp: package `homology' not installed or its binaries are not available, falling back to (slower) internal homology algorithms.
```

**Comment by mansurzubairu** [comment]
packages with binaries like the Semigroups packages should be wrapped and be working on windows users.

**Comment by alex-konovalov** [comment]
Ticked the box for smallsemi - its new release  is just published.

**Comment by alex-konovalov** [comment]
Thanks to @hulpke for the new fix for SmallGrp in https://github.com/gap-packages/smallgrp/pull/46 - to be reviewed and released in time for GAP 4.11

**Comment by alex-konovalov** [comment]
This issue is replaced by #3995 tracking progress towards packages for GAP 4.12.0
